workers/unity/Packages/com.improbable.gdk.objectpooling/ObjectPooler.cs

```diff
@@ -11,13 +11,26 @@ public class ObjectPooler : MonoBehaviour
 
         private void Awake()
         {
+            // Destroy the ObjectPooler component being created if an Instance already exists
+            if (Instance != null)
+            {
+                Destroy(this);
+                return;
+            }
+
             Instance = this;
             objectPools = new Dictionary<GameObject, IObjectPool>();
         }
 
         public static ObjectPool<T> GetOrCreateObjectPool<T>(GameObject prefab, int prePoolCount = 0)
             where T : IPoolableObject
         {
+            if (Instance == null)
+            {
+                // Creation of the ObjectPooler component sets the Instance field
+                new GameObject("ObjectPool").AddComponent<ObjectPooler>();
```

Comment:
Given this is basically a singleton GameObject, which doesn't really have any reason to be a GameObject.. can't this just be made a classic singleton class?

Reply:
Generally good practice to have unused instances of the pool nested under the pool to keep the root of the hierarchy clean.
(Not sure if that's currently happening, but I'd say that'd be a nicer change than removing the gameObject entirely)

Reply:
Sure that makes sense.

I do recall hearing that keeping the objects in an nested hierarchy is worse for performance (although likely negligible here) since any change in a child object forces a re-evaluation of all the objects in its tree.

Reply:
It could be - but having a GameObject for it lets us place all the pooled objects as children under it